process/process.go

```diff
@@ -40,25 +40,33 @@ type Creator interface {
 	New(ctx context.Context, args ...string) (Process, error)
 }
 
+type CmdCreatorFunc func(ctx context.Context, exePath string, args ...string) (*exec.Cmd, error)
+
 type exeProcessCreator struct {
 	exePath string
+	cmdCreatorFunc CmdCreatorFunc
+}
+
+func NewCmd(ctx context.Context, exePath string, args ...string) (*exec.Cmd, error) {
+	cmd := exec.CommandContext(ctx, exePath, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd, nil
 }
 
 // NewCreator creates a Creator for external Tor process execution based on the
 // given exe path.
-func NewCreator(exePath string) Creator {
-	return &exeProcessCreator{exePath}
+func NewCreator(exePath string, newCmd CmdCreatorFunc) Creator {
```

Comment:
This breaks backwards compatibility. Rather, I was thinking of having `type CmdCreatorFunc func(ctx context.Context, args ...string) (*exec.Cmd, error)` and then have that implement `Creator` via something like `func (c CmdCreatorFunc) New(ctx context.Context, args ...string) (Process, error) { /* send back error or &exeProcess after calling c(ctx, args) */ }`. And leave the rest of the code alone (or even better, change `NewCreator` to create and return a `CmdCreatorFunc`).

Sorry I'm too busy to do this myself, I hope that's clear.